[Security Solution] Show deprecated bulk endpoints in Upgrade Assistant (#207091)
elastic/kibana: docs/upgrade-notes.asciidoc

[float]
=== Breaking changes

[discrete]
[[breaking-207091]]
.Removed legacy security rules bulk endpoints (9.0.0)
[%collapsible]
====
*Details* +
--
* `POST /api/detection_engine/rules/_bulk_create` has been replaced by `POST /api/detection_engine/rules/_import`
* `PUT /api/detection_engine/rules/_bulk_update` has been replaced by `POST /api/detection_engine/rules/_bulk_action`
* `PATCH /api/detection_engine/rules/_bulk_update has been replaced by `POST /api/detection_engine/rules/_bulk_action`
* `DELETE /api/detection_engine/rules/_bulk_delete` has been replaced by `POST /api/detection_engine/rules/_bulk_action`
* `POST api/detection_engine/rules/_bulk_delete` has been replaced by `POST /api/detection_engine/rules/_bulk_action`
--

> These changes were introduced in PR {kibana-pull}#197422[#197422]

Reviewer: Made a few minor changes:
- Removed the hash symbol as it might be messing up the link. The link leads users to [this page](https://github.com/elastic/kibana/pulls#197422) instead of https://github.com/elastic/kibana/pull/197422. 
- Removed "PR" because users can see they're viewing a PR if they click the link.
- Added a period at the end since this is a complete sentence.  

```suggestion
These changes were introduced in {kibana-pull}197422[#197422].
```

*Impact* +
Deprecated endpoints will fail with a 404 status code starting from version 9.0.0

*Action* +
--
Update your implementations to use the new endpoints:

> 1. **For bulk creation of rules:**

Reviewer: Since users don't need to follow numbers 1-3 in order, an unordered list is more appropriate here.
 
```suggestion
* **For bulk creation of rules:**
```

>    - Use `POST /api/detection_engine/rules/_import` (link:https://www.elastic.co/docs/api/doc/kibana/v8/operation/operation-importrules[API documentation]) to create multiple rules along with their associated entities (e.g., exceptions and action connectors).

Reviewer: @natasha-moore-elastic I need a sanity check here (please and thank you). The link in the preview improperly resolves to https://www.elastic.co/docs/api/doc/kibana/v8/operation/operation-update-fleet-proxies, instead of the URL above. Do you know why this is happening? Here's what I'm seeing:

![operation-update-fleet-proxies-url](https://github.com/user-attachments/assets/bb583663-5cb2-4d09-9dd3-b6f774f31a6d)

Reviewer: ```suggestion
   - Use `POST /api/detection_engine/rules/_import` (link:https://www.elastic.co/docs/api/doc/kibana/v8/operation/operation-importrules[API documentation]) to create multiple rules along with their associated entities (for example, exceptions and action connectors).
```

Reviewer: I checked in incognito mode, and the link resolves properly (whew!). You're good, @nikitaindik! 

Also, 1+ to @natasha-moore-elastic's suggestion about using the `{api-kibana}` attribute instead of hard-coding the site's URL.

>    - Alternatively, create rules individually using `POST /api/detection_engine/rules` (link:https://www.elastic.co/docs/api/doc/kibana/v8/operation/operation-createrule[API documentation]).

Reviewer: @natasha-moore-elastic this link is also resolving improperly. The URL in the file looks good, but the link in the preview leads you to https://www.elastic.co/docs/api/doc/kibana/v8/operation/operation-update-package-policy.

> 2. **For bulk updates of rules:**

Reviewer: ```suggestion
* **For bulk updates of rules:**
```

   - Use `POST /api/detection_engine/rules/_bulk_action` (link:https://www.elastic.co/docs/api/doc/kibana/v8/operation/operation-performrulesbulkaction[API documentation]) to update fields in multiple rules simultaneously.
   - Alternatively, update rules individually using `PUT /api/detection_engine/rules` (link:https://www.elastic.co/docs/api/doc/kibana/v8/operation/operation-updaterule[API documentation]).

> 3. **For bulk deletion of rules:**

Reviewer: ```suggestion
* **For bulk deletion of rules:**
```

>    - Use `POST /api/detection_engine/rules/_bulk_action` (link:https://www.elastic.co/docs/api/doc/kibana/v8/operation/operation-performrulesbulkaction[API documentation]) to delete multiple rules by IDs or query.

Reviewer: @natasha-moore-elastic this link is improperly resolving to https://www.elastic.co/docs/api/doc/kibana/v8/operation/operation-post-delete-package-policy in the preview.

Reviewer: Same as above.

>    - Alternatively, delete rules individually using `DELETE /api/detection_engine/rules` (link:https://www.elastic.co/docs/api/doc/kibana/v8/operation/operation-deleterule[API documentation]).

Reviewer: @natasha-moore-elastic this link is improperly resolving to https://www.elastic.co/docs/api/doc/kibana/v8/operation/operation-update-package-policy in the preview.

Reviewer: Same as above.
